src/util/imagefiledata.cpp

```diff
@@ -0,0 +1,26 @@
+#include "util/imagefiledata.h"
+
+#include <QByteArray>
+#include <QDebug>
+#include <QFile>
+#include <QFileInfo>
+#include <QImage>
+
+ImageFileData::ImageFileData(const QString& coverArtFilePath)
+        : QImage(coverArtFilePath) {
+    m_coverArtExtension = QFileInfo(coverArtFilePath).suffix();
+    QFile coverArtFile = QFile(coverArtFilePath);
+    coverArtFile.open(QIODevice::ReadOnly);
+    m_coverArtBytes = coverArtFile.readAll();
+    coverArtFile.close();
+}
+
+// Most of the cover art extensions are 'jpg' at cover art archive.
+ImageFileData::ImageFileData(const QByteArray& coverArtBytes)
```

Comment:
This should become: 
ImageFileData::ImageFileData(const QByteArray &data, const char *format = nullptr)

Reply:
Again, I am just asking for to learn, this is for the extension of the cover art?

Reply:
do we know the format in case of web download, to populate m_coverArtExtension? 
Else we may copy:
https://github.com/qt/qtbase/blob/ee3ac3a3bfb4c4a0802f6530f61995297c573646/src/gui/image/qimage.cpp#L3485
here and use ``QImageReader::imageFormat(QIODevice *device)``` to request the extensions form the file content.

Reply:
most of the cover arts that I've seen were `jpg`. Do we still need the extension. If we need it, does it need to be static?